### PR TITLE
Update plugin API

### PR DESCRIPTION
**Pull request by yann300**
following https://github.com/ethereum/remix-ide/issues/1156

### CODE REVIEW

File: src/app/plugin/pluginManager.js [hunk 1]
```diff
@@ -61,37 +101,47 @@ class PluginManager {
...
         var data = JSON.parse(event.data)
-        data.arguments.unshift(this.inFocus)
+        data.value.unshift(this.inFocus)
         if (allowedapi[data.type]) {
-          data.arguments.push((error, result) => {
-            response(data.type, data.id, error, result)
+          data.value.push((error, result) => {
+            response(data.key, data.type, data.id, error, result)
           })
           api[data.key][data.type].apply({}, data.arguments)
```
Comment (Fi3) [inline src/app/plugin/pluginManager.js]: `data.arguments` should be `data.value`

File: test-browser/plugin/remix.js [hunk 2]
```diff
@@ -8,25 +8,31 @@ window.addEventListener('message', receiveMessage, false)
 window.onload = function () {
   document.querySelector('input#testmessageadd').addEventListener('click', function () {
     window.parent.postMessage(JSON.stringify({
+      action: 'request',
+      key: 'editor',
```
Comment (Fi3) [inline test-browser/plugin/remix.js]: `key: 'editor'` should be `key: 'config'`

File: test-browser/plugin/remix.js [hunk 3]
```diff
@@ -8,25 +8,31 @@ window.addEventListener('message', receiveMessage, false)
...
+      value: [document.getElementById('filename').value, document.getElementById('valuetosend').value],
       id: 34
-    }), 'http://127.0.0.1:8080')
+    }), '*')
   })
 
   document.querySelector('input#testmessageremove').addEventListener('click', function () {
     window.parent.postMessage(JSON.stringify({
+      action: 'request',
+      key: 'editor',
```
Comment (Fi3) [inline test-browser/plugin/remix.js]: `key: 'editor'` should be `key: 'config'`

File: test-browser/plugin/remix.js [hunk 4]
```diff
@@ -8,25 +8,31 @@ window.addEventListener('message', receiveMessage, false)
...
+      value: [document.getElementById('filename').value],
       id: 35
-    }), 'http://127.0.0.1:8080')
+    }), '*')
   })
 
   document.querySelector('input#testmessagerget').addEventListener('click', function () {
     window.parent.postMessage(JSON.stringify({
+      action: 'request',
+      key: 'editor',
```
Comment (Fi3) [inline test-browser/plugin/remix.js]: `key: 'editor'` should be `key: 'config'`

File: src/app/plugin/pluginManager.js [hunk 5]
```diff
@@ -1,13 +1,47 @@
...
+ *
+ * - Response sent by Remix and receive by the plugin:
+ *{
+ *  id: <number>,
+ *  action: 'response',
+ *  key: <string>,
+ *  type: <string>,
+ *  value: <array>,
+ *  error: (see below)
+ *}
```
Comment (serapath) [inline src/app/plugin/pluginManager.js]: I just checked again: https://github.com/ethereum/remix-ide/issues/1156

* The original proposal is `var { id, status, type, key, value } = msg`
* You said you would like to add the `action` field to indicate "response/notification/request".
* That would make it `var { action, id, status, type, key, value } = msg`
* But now the `status` is/was renamed to `error` i guess.

Just to make sure how i have to think about it :-)
So i guess we now have: `var { action, id, error: status, type, key, value } = msg`

I would say that is fine.

Comment (serapath) [reply src/app/plugin/pluginManager.js]: The above comment is just for me to clarify.
If this is how it's meant, i would say the PR looks good to me :-)

Comment (yann300) [reply src/app/plugin/pluginManager.js]: yes except that error is undefined if there's no error

**Pull request merged**